fastlane/lib/fastlane/runner.rb

```diff
@@ -228,7 +228,11 @@ def execute_action(method_sym, class_ref, arguments, custom_dir: nil, from_actio
           # If another action is calling this action, we shouldn't show it in the summary
           # (see https://github.com/fastlane/fastlane/issues/4546)
 
-          action_name = from_action ? nil : class_ref.step_text
+          if arguments.kind_of?(Array) && arguments.first.kind_of?(Hash)
+            step_name_override = arguments.first[:step_name]
+            arguments.first.delete(:step_name)
+          end
+          action_name = from_action ? nil : (step_name_override || class_ref.step_text)
```

Comment:
I find this confusing to read (not your fault, part of it's the original implementation)

What about something like this:

```ruby
unless from_action # see https://github.com/fastlane/fastlane/issues/4546
  action_name = arguments.first[:step_name] || class_ref.step_text
end
```

I think you can remove `if arguments.kind_of?(Array) && arguments.first.kind_of?(Hash)` by either

- moving the `arguments = [{}] if arguments.empty?` on line 137 out of its conditional
- moving the `if arguments.count == 0` stuff below to above

Reply:
I do agree :)

Reply:
I did not manage to get ride of the kind_of check by trying the two alternative.
the first alternative break many test assumptions